miden/tests/integration/helpers.rs

```diff
@@ -1,11 +1,25 @@
-use processor::{ExecutionError, ExecutionTrace, Process};
+use processor::{ExecutionError, ExecutionTrace, Process, VmStateIterator};
 use proptest::prelude::*;
 pub use vm_core::{program::Script, Felt, FieldElement, ProgramInputs, MIN_STACK_DEPTH};
 
 // CONSTANTS
 // ================================================================================================
 pub const WORD_LEN: usize = 4;
 
+#[test]
+fn simple_exec_iter() {
+    let test = Test::new("begin push.1 push.2 add end");
+    let states = test.execute_iter();
+    let mut i = 0;
+    for state in states {
+        let result = state.as_ref();
+        assert!(!result.is_err());
+        let state = result.unwrap();
+        assert_eq!(i, state.clk);
+        i += 1;
+    }
+}
```

Comment:
We probably shouldn't put an actual test in the test helper file.

Reply:
Sorry this got moved around after all the changes, just removed the test